packages/core/src/util/transactionSummaryInspector.ts

```diff
@@ -0,0 +1,242 @@
+  certificates: Cardano.Certificate[],
+  rewardAccounts: Cardano.RewardAccount[],
+  poolIds: Set<Cardano.PoolId>,
+  networkId: Cardano.NetworkId
+): { deposit: Cardano.Lovelace; returnedDeposit: Cardano.Lovelace } => {
+  let deposit = 0n;
+  let returnedDeposit = 0n;
+
+  // TODO: For the case of deregistration (StakeDeregistration and PoolRetirement) the code here is not entirely correct
+  // as we are assuming the current protocol parameters for the deposits where the same as the ones used when the certificates where issued.
+  // This is going to work for now, but to properly implement this we need a way to know when the certificate we are undoing was originally issued
+  // and get the protocol parameters for that epoch. However, these parameters in particular have never change in mainnet, so this is probably
+  // is good for now.
+  for (const cert of certificates) {
+    switch (cert.__typename) {
+      case Cardano.CertificateType.StakeRegistration:
+        if (rewardAccounts.includes(Cardano.RewardAccount.fromCredential(cert.stakeCredential, networkId)))
+          deposit += depositParams.stakeKeyDeposit;
+        break;
+      case Cardano.CertificateType.StakeDeregistration:
+        if (rewardAccounts.includes(Cardano.RewardAccount.fromCredential(cert.stakeCredential, networkId)))
+          returnedDeposit += depositParams.stakeKeyDeposit;
+        break;
+      case Cardano.CertificateType.PoolRegistration:
+        if (rewardAccounts.some((acct) => cert.poolParameters.owners.includes(acct)))
+          deposit += depositParams.poolDeposit;
+        break;
+      case Cardano.CertificateType.PoolRetirement: {
+        if (poolIds.has(cert.poolId)) returnedDeposit += depositParams.poolDeposit;
+        break;
+      }
+    }
+  }
+
+  return {
+    deposit,
+    returnedDeposit
+  };
+};
+
+const computeConwayDeposits = (
+  certificates: Cardano.Certificate[],
+  rewardAccounts: Cardano.RewardAccount[],
+  networkId: Cardano.NetworkId
+): { deposit: Cardano.Lovelace; returnedDeposit: Cardano.Lovelace } => {
+  let deposit = 0n;
+  let returnedDeposit = 0n;
+
+  for (const cert of certificates) {
+    switch (cert.__typename) {
+      case Cardano.CertificateType.Registration:
+      case Cardano.CertificateType.StakeRegistrationDelegation:
+      case Cardano.CertificateType.VoteRegistrationDelegation:
+      case Cardano.CertificateType.StakeVoteRegistrationDelegation:
+        if (rewardAccounts.includes(Cardano.RewardAccount.fromCredential(cert.stakeCredential, networkId)))
+          deposit += cert.deposit;
+        break;
+      case Cardano.CertificateType.Unregistration:
+        if (rewardAccounts.includes(Cardano.RewardAccount.fromCredential(cert.stakeCredential, networkId)))
+          returnedDeposit += cert.deposit;
+        break;
+    }
+  }
+
+  return {
+    deposit,
+    returnedDeposit
+  };
+};
+
+/** Inspects a transaction for its deposits and returned deposits. */
+export const getDeposits = (
```

Comment:
Consider consolidating this with `computeImplicitCoin`. Perhaps adding an optional `rewardAccounts` argument to it rather than reimplementing the computation entirely would be better?

Reply:
I thought about this, but in the end I re-implemented the code because it is a bit different, computeImplicitCoin assumes all certificates belong to the wallet (I guess is because its only ever internally used by the PersonalWallet during input selection for transaction its building) and it consolidates the return deposits with the withdrawals (although it can be easily reversed with a subtraction). 

I will still need to have separate logic for deposits calculation if we make rewardAddresses optional. I can move the getDeposits function into the util and call that if the rewardAddresses are provided, and otherwise, call the original logic (but now that I look at it closely it probably also needs to be updated to work with conway certificates). 

Alternatively we can make the util take a mandatory rewardAddresses array rather than an optional one and we keep the logic the same for both cases. Let me know what you think (adding an extra field "returnedDeposits" would be also convenient)

Reply:
Returning `returnedDeposits` (we call it 'deposit reclaim' elsewhere in code) from `computeImplicitCoin` would be good. However Making `rewardAddresses` mandatory wouldn't work for input selection, so you'd have to do some conditional logic there with optional addresses. I don't like having 2 utilities that do (pretty much) the same thing here. I think we should either:
- hoist `computeImplicitCoin` to `input-selection` package so that it's more explicitly scoped for input selection usage
- consolidate the 2 in core package and do conditional logic based on presence of rewardAddresses param. Even if implementation is different based on parameters, there would be some reuse, and more importantly there would be a single function to maintain

Reply:
done here @mkazlauskas https://github.com/input-output-hk/cardano-js-sdk/pull/1057/commits/293cb9a7df5fb1c52ada864c5ab4e19c606c3ce1